uaclient/timer/update_messaging.py

```diff
@@ -125,17 +127,20 @@ def update_motd_messages(cfg: UAConfig) -> bool:
         service = "n/a"
         pkg_num = 0
 
+        enabled_services = _enabled_services(cfg).enabled_services
         if system.is_current_series_active_esm():
-            esm_infra_status, _ = ESMInfraEntitlement(cfg).application_status()
-            if esm_infra_status == ApplicationStatus.ENABLED:
-                service = "esm-infra"
+            service = "esm-infra"
+            if any(
+                ent.name for ent in enabled_services if ent.name == service
+            ):
```

Comment:
I don't disagree with these types of refactors, but they're not necessary for the main machine_token_file refactor AFAICT.

If you want to keep this, please simplify it to:
```suggestion
if any(ent.name == "esm-infra" for ent in enabled_services):
```